### PR TITLE
Use bindings-DSL, drop bindings-posix dependency

### PR DESCRIPTION
**Pull request by phadej**
*EDIT* update: though it compiles on windows, and I even made tests compile on windows (had to edit a bit more), I cannot make test pass :/ (yet)

### CODE REVIEW

**Comment by klao** [comment]
Thanks a lot for this work!

Because these are used only for benchmarks, it's not very important, so I'm just going to drop the dependencies. Both bindings-posix (I have a hand-roller `localtime_r` binding anyway) and `timezone-{series,olson}` (at least for now).

**Comment by phadej** [comment]
`c'tzset` is used in tests too..

File: stack.yaml [hunk 1]
```diff
@@ -6,28 +6,22 @@ resolver: nightly-2016-05-29
...
+    commit: f8dece8c016db6476e2bb0d4f972769a76f6ff40
+  extra-dep: true
+
+# For timezone-olson
+allow-newer: true
 
 # Packages to be pulled from upstream that are not in the resolver (e.g., acme-missiles-0.3)
 extra-deps:
-- bindings-posix-1.2.6
+- thyme-0.3.5.5
```
Comment (klao) [inline stack.yaml]: Quick question: why did you need this? That's the version for thyme in the nightly, or am I seeing something wrong?

Comment (phadej) [inline stack.yaml]: Probably I added it before I bumped the `resolver:`. No good reason.

**Comment by klao** [comment]
I ended up keeping the `timezone-series` benchmarks and just removed everything depending on `bindings-posix`.

With this, I am going to close this pull request. Thanks again for all the help!

Oh, I am going to try to test it on Windows, but if you send me additional details, that's always welcome!
